src/Client.php

```diff
@@ -39,20 +39,27 @@ class AvaTaxClientBase
     private $environment;
 
     /**
-     * Construct a new AvaTaxClient 
+     * @var bool        The setting for whether the client should catch exceptions
+     */
+    private $catchExceptions;
+
+    /**
+     * Construct a new AvaTaxClient
      *
-     * @param string $appName      Specify the name of your application here.  Should not contain any semicolons.
-     * @param string $appVersion   Specify the version number of your application here.  Should not contain any semicolons.
-     * @param string $machineName  Specify the machine name of the machine on which this code is executing here.  Should not contain any semicolons.
-     * @param string $environment  Indicates which server to use; acceptable values are "sandbox" or "production", or the full URL of your AvaTax instance.
-     * @param array $guzzleParams  Extra parameters to pass to the guzzle HTTP client (http://docs.guzzlephp.org/en/latest/request-options.html)
+     * @param string $appName       Specify the name of your application here.  Should not contain any semicolons.
+     * @param string $appVersion    Specify the version number of your application here.  Should not contain any semicolons.
+     * @param string $machineName   Specify the machine name of the machine on which this code is executing here.  Should not contain any semicolons.
+     * @param string $environment   Indicates which server to use; acceptable values are "sandbox" or "production", or the full URL of your AvaTax instance.
+     * @param array $guzzleParams   Extra parameters to pass to the guzzle HTTP client (http://docs.guzzlephp.org/en/latest/request-options.html)
+     * @param bool $catchExceptions Indicates if client exceptions should automatically be caught and returned as a message string
      */
-    public function __construct($appName, $appVersion, $machineName, $environment, $guzzleParams = [])
+    public function __construct($appName, $appVersion, $machineName, $environment, $guzzleParams = [], $catchExceptions = true)
```

Comment:
This looks like a great change - but can we please move this out of the constructor and move it into a `withCatchExceptions()` builder function?

Reply:
That sounds like a good call. I have updated the feature so that the property can now only be set through `withCatchExceptions()`.

Please let me know if you see any other improvements that could be made.